java/statestore/src/test/java/sleeper/statestore/s3/S3StateStoreIT.java

```diff
@@ -1201,38 +1075,19 @@ public void shouldThrowExceptionWithPartitionSplitRequestWhereParentIsMarkedAsLe
         Field field = new Field("key", new LongType());
         Schema schema = Schema.builder().rowKeyFields(field).build();
         StateStore dynamoDBStateStore = getStateStore(schema);
-        Partition parentPartition = dynamoDBStateStore.getAllPartitions().get(0);
-        Partition parentPartitionAfterSplit = parentPartition.toBuilder()
-                .childPartitionIds(Arrays.asList("child1", "child2"))
-                .leafPartition(true)
-                .build();
-        Region region1 = new Region(new RangeFactory(schema).createRange(field, Long.MIN_VALUE, null));
-        Partition childPartition1 = Partition.builder()
-                .rowKeyTypes(new LongType())
-                .leafPartition(true)
-                .id("child1")
-                .region(region1)
-                .childPartitionIds(new ArrayList<>())
-                .parentPartitionId("parent")
-                .build();
-        Region region2 = new Region(new RangeFactory(schema).createRange(field, Long.MIN_VALUE, null));
-        Partition childPartition2 = Partition.builder()
-                .rowKeyTypes(new LongType())
-                .leafPartition(true)
-                .id("child2")
-                .region(region2)
-                .childPartitionIds(new ArrayList<>())
-                .parentPartitionId("parent")
-                .build();
-
+        PartitionTree tree = new PartitionsBuilder(schema)
+                .rootFirst("root")
+                .splitToNewChildren("root", "child1", "child2", Long.MIN_VALUE)
+                .buildTree();
+        dynamoDBStateStore.atomicallyUpdatePartitionAndCreateNewOnes(tree.getPartition("root"), tree.getPartition("child1"), tree.getPartition("child2"));
```

Comment:
This doesn't match the behaviour of the test before. The test is intended to check that an update will fail to validate if the leafPartition flag is set incorrectly on the parent partition after the split.

The test was invalid before, but we should probably actually implement it now.